### PR TITLE
FINAL: SC3-1-1-html-lang-valid

### PR DESCRIPTION
**Pull request by jkodu**
- Implement as rule from draft - https://github.com/auto-wcag/auto-wcag/pull/80
- Implementation tests are auto generated from the `rule.html` liquid template, as shown in this PR - https://github.com/auto-wcag/auto-wcag/pull/107/files
- Fixes issue #165

### CODE REVIEW

**Comment by annethyme** [comment]
Very editorial comment for the Description: 
consider adding "that" in "This rule checks the lang or xml:lang attribute conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt)." --> "This rule checks that the lang or xml:lang attribute conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt)."

**Comment by WilcoFiers** [comment]
@jeankaplansky to review

File: _rules/SC3-1-1-html-lang-valid.md [hunk 1]
```diff
@@ -19,13 +19,13 @@ authors:
...
 
-The root element of the page, if it is an `html` element with a `lang` and/or `xml:lang` attribute.
+The root element of the page, if it is an `html` element with a [non-empty](#non-empty) `lang` and/or `xml:lang` attribute.
 
 ### Expectation
 
 The value of the `lang` or `xml:lang` attribute conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
 
-**Note**: If `lang` and `xml:lang` exists, then only `lang` is tested for conformance.
+**Note**: If both `lang` and `xml:lang` exists, then only `lang` is tested for conformance.
```
Comment (jeankaplansky) [inline _rules/SC3-1-1-html-lang-valid.md]: In the event of parser presence - what takes precedence - xml:lang is expected - but this is polymorph HTML5 if xml:lang is present at all which implies that a parser IS present. @jeankaplansky to investigate from the XML parser context. This applies to https://github.com/auto-wcag/auto-wcag/pull/107, https://github.com/auto-wcag/auto-wcag/pull/109, and https://github.com/auto-wcag/auto-wcag/pull/96

File: _rules/SC3-1-1-html-lang-valid.md [hunk 2]
```diff
@@ -19,13 +19,13 @@ authors:
 
 ### Applicability
 
-The root element of the page, if it is an `html` element with a `lang` and/or `xml:lang` attribute.
+The root element of the page, if it is an `html` element with a [non-empty](#non-empty) `lang` and/or `xml:lang` attribute.
 
 ### Expectation
 
 The value of the `lang` or `xml:lang` attribute conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).
```
Comment (carlosapaduarte) [inline _rules/SC3-1-1-html-lang-valid.md]: Editorial: Consider removing the **Note** and writing the Expectation as
The value of the `lang` attribute, or the `xml:lang` attribute if the `lang` attribute does not exist, conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).

File: _rules/SC3-1-1-html-lang-valid.md [hunk 3]
```diff
@@ -43,6 +43,54 @@ There are known combinations of a popular operating system with browsers and ass
...
+
+```html
+<html xml:lang="fr" lang="fr">
+```
+
+```html
+<html lang="fr" xml:lang="xyz">
+```
+
+## Failed
```
Comment (carlosapaduarte) [inline _rules/SC3-1-1-html-lang-valid.md]: Include a test case similar to the fourth Pass
+```html
+<html lang="xyz" xml:lang="fr">
+```

File: _rules/SC3-1-1-html-lang-valid.md (hunk 2 above)
Comment (carlosapaduarte) [inline _rules/SC3-1-1-html-lang-valid.md]: Editorial: Consider removing the **Note** and writing the Expectation as:
The value of the `lang` attribute, or the `xml:lang` attribute if the `lang` attribute does not exist, conforms to [BCP 47](https://www.ietf.org/rfc/bcp/bcp47.txt).

**Pull request merged**